### PR TITLE
Add ContentSource table and Content lifecycle integration test

### PR DESCRIPTION
**Pull request by fearnworks**
- Add ContentSource table, schema, crud, and endpoints 
- Update existing Content logic to use ContentSource
- Add 4 integration smoke tests to test Create, Read, Update, and Delete of content

### CODE REVIEW

File: modules/odr_api/tests/integration/models/test_content_lifecycle.py
Comment (Tomislaw) [inline modules/odr_api/tests/integration/models/test_content_lifecycle.py:1]: Do you plan to add negative test cases (such as trying to remove content without permissions, trying to update missing content, or adding content whose md5 hash already exist, etc)

Ot it isn't scope of this task?

Comment (fearnworks) [reply modules/odr_api/tests/integration/models/test_content_lifecycle.py:1]: I'm not opposed, and we certainly will want this for more continuous cases. But I think for the initial tests we can be fine without it. Mainly targeting a "does it work at all" validation

File: modules/odr_core/odr_core/models/content.py [hunk 1]
```diff
@@ -56,8 +76,22 @@ class ContentAuthor(Base):
...
     content = relationship("Content", back_populates="content_authors")
+
+
+class ContentSource(Base):
+    __tablename__ = "content_sources"
+
+    id = Column(Integer, primary_key=True, index=True)
+    content_id = Column(Integer, ForeignKey("contents.id"))
+    type = Column(Enum(ContentSourceType))
+    value = Column(String)
```
Comment (Tomislaw) [inline modules/odr_core/odr_core/models/content.py:92]: We should make 'value' unique, as we don't expect to have multiple ContentSources with same address

File: modules/odr_core/odr_core/crud/content.py [hunk 2]
```diff
@@ -59,21 +151,42 @@ def update_content(db: Session, content_id: int, content: ContentUpdate) -> Opti
 def delete_content(db: Session, content_id: int) -> bool:
     db_content = db.query(Content).filter(Content.id == content_id).first()
     if db_content:
+        # Delete associated sources
+        for source in get_content_sources(db, content_id):
```
Comment (Tomislaw) [inline modules/odr_core/odr_core/crud/content.py:155]: I would use ondelete='CASCADE' / cascade='all,delete-orphan' instead of removing it in loop

**Review by Tomislaw — CHANGES_REQUESTED** [review]

File: modules/odr_core/odr_core/crud/content.py (hunk 2 above)
Comment (fearnworks) [reply modules/odr_core/odr_core/crud/content.py:155]: good put

**Review by Tomislaw — APPROVED** [review]

**Pull request merged**